source/extensions/path/uri_template_lib/uri_template_internal.h

```diff
@@ -47,91 +49,101 @@ using ParsedSegment = absl::variant<Operator, Variable, Literal>;
 /**
  * Represents the parsed path including literals and variables.
  */
-struct ParsedUrlPattern {
+struct ParsedPathPattern {
   std::vector<ParsedSegment> parsed_segments_;
+
+  // Suffix holds end of path that matched a wildcard.
+  // For example:
+  // Pattern: /foo/bar/**
+  // Path: /foo/bar/some/more/stuff
+  // Suffix: some/more/stuff
   absl::optional<absl::string_view> suffix_;
   absl::flat_hash_set<absl::string_view> captured_variables_;
 
   std::string debugString() const;
 };
 
 /**
- * Check if literal is valid
- * Literals cannot hold wildcards or curl brackets.
+ * Returns true if `literal` is valid for pattern match.
+ * (does not contain wildcards or curly brackets).
  */
 bool isValidLiteral(absl::string_view literal);
 
 /**
- * Check if rewrite literal is valid
- * Literals cannot hold wildcards or curl brackets.
+ * Returns true if `literal` is valid for pattern rewrite.
+ * (does not contain wildcards or curly brackets).
```

Comment:
This comment appears identical to the previous, but presumably these two method are not the same so the comment should make clear what the difference is.

Reply:
One comment says `pattern match` the other `pattern rewrite`

Reply:
Sorry, I meant the "(does not contain any wildcards or curly brackets)" part